src/controllers/serviceproviders.js

```diff
@@ -0,0 +1,149 @@
+import ServiceProviders from "../models/ServiceProviders.js";
+
+export const createServiceProvider = async (req, res) => {
+  try {
+    const providers = await ServiceProviders.find().populate("appointments");
+    const newProvider = await ServiceProviders.create({
+      ...req.body,
+      isActive: true,
+    });
+    const isProviderRegister = providers.some(
+      (prov) => prov.email === newProvider.email
+    );
+    if (isProviderRegister) {
+      return res.status(451).json({
+        message: "This email has already register",
+        error: true,
+        data: req.body,
+      });
+    }
+    await newProvider.save();
+    return res.status(201).json({
+      message: `Service provider created successfully`,
+      data: newProvider,
+      error: false,
+    });
+  } catch (error) {
+    res.status(500).json({
+      message: `Unespected error ${error}.`,
+      error: true,
+      data: undefined,
+    });
+  }
+};
+
+export const getProviders = async (req, res) => {
+  try {
+    const providerList = await ServiceProviders.find().populate("appointments");
+    if (providerList.length === 0) {
+      return res.status(404).json({
+        mesage: "Service Provider list is empty. Register one first!",
+        error: true,
+        data: undefined,
+      });
+    }
+    return res.status(200).json({
+      message: "Service Providers found!",
+      error: false,
+      data: [...providerList],
+    });
+  } catch (error) {
+    res.status(500).json({
+      message: `Unespected error ${error}.`,
+      error: true,
+      data: undefined,
+    });
+  }
+};
+
+export const getOneProvider = async (req, res) => {
+  try {
+    const { id } = req.body;
+    const provider = await ServiceProviders.findById(id).populate(
```

Comment:
Here you can add a validation that checks that the id sent as path params is a valid id using: 

`function isValidObjectId(stringToCheck) {
    return mongoose.Types.ObjectId.isValid(stringToCheck);
}`